test/libraries/SafeMigration.spec.ts

```diff
@@ -0,0 +1,344 @@
+            safeL2RuntimeBytecode: hre.artifacts.readArtifactSync("SafeL2").deployedBytecode,
+            fallbackHandlerRuntimeBytecode: hre.artifacts.readArtifactSync("CompatibilityFallbackHandler").deployedBytecode,
+        },
+    },
+    {
+        testSuiteName: "1.3.0 to 1.4.1",
+        from: { safeDeploymentData: deploymentData.safe130, safeL2DeploymentData: deploymentData.safe130l2 },
+        to: {
+            safeRuntimeBytecode: safeRuntimeBytecode.safe141,
+            safeL2RuntimeBytecode: safeRuntimeBytecode.safe141l2,
+            fallbackHandlerRuntimeBytecode: safeRuntimeBytecode.safe141fallbackHandler,
+        },
+    },
+    {
+        testSuiteName: "1.4.1 to latest (1.5.0)",
+        from: { safeDeploymentData: deploymentData.safe141, safeL2DeploymentData: deploymentData.safe141l2 },
+        to: {
+            safeRuntimeBytecode: hre.artifacts.readArtifactSync("Safe").deployedBytecode,
+            safeL2RuntimeBytecode: hre.artifacts.readArtifactSync("SafeL2").deployedBytecode,
+            fallbackHandlerRuntimeBytecode: hre.artifacts.readArtifactSync("CompatibilityFallbackHandler").deployedBytecode,
+        },
+    },
+];
+
+describe("SafeMigration Library", () => {
+    const migratedInterface = new ethers.Interface(["function masterCopy() view returns(address)"]);
+
+    migrationPaths.forEach(({ testSuiteName, from, to }) => {
+        describe(testSuiteName, () => {
+            const setupTests = deployments.createFixture(async ({ deployments }) => {
+                await deployments.fixture();
+                const signers = await ethers.getSigners();
+                const [user1] = signers;
+
+                // Set the runtime code for hardcoded addresses, so the expected events are emitted
+                await hre.network.provider.send("hardhat_setCode", [SAFE_SINGLETON_ADDRESS, to.safeRuntimeBytecode]);
+                await hre.network.provider.send("hardhat_setCode", [SAFE_SINGLETON_L2_ADDRESS, to.safeL2RuntimeBytecode]);
+                await hre.network.provider.send("hardhat_setCode", [COMPATIBILITY_FALLBACK_HANDLER, to.fallbackHandlerRuntimeBytecode]);
+
+                const singletonAddress = (await (await user1.sendTransaction({ data: from.safeDeploymentData })).wait())?.contractAddress;
+                const singletonL2Address = (await (await user1.sendTransaction({ data: from.safeL2DeploymentData })).wait())
+                    ?.contractAddress;
+
+                if (!singletonAddress || !singletonL2Address) {
+                    throw new Error("Could not deploy safe or safeL2");
+                }
+                const singleton = await getSafeSingletonAt(singletonAddress);
+                const singletonL2 = await getSafeSingletonAt(singletonL2Address);
+
+                const migration = await (
+                    await safeMigrationContract()
+                ).deploy(SAFE_SINGLETON_ADDRESS, SAFE_SINGLETON_L2_ADDRESS, COMPATIBILITY_FALLBACK_HANDLER);
```

Comment:
This will deploy a new contracts and not test the actual deployment from the deployment script.

Reply:
https://github.com/safe-global/safe-smart-account/pull/793/commits/0b53296d0ad01d16b32b7dc07381239306429911#diff-3aa3f08a1342c64e095f5ed125dc4b4f2bc28e89a8bc9a90a9ff4f85efeb0ece